### PR TITLE
Refactor/api calls

### PR DESCRIPTION
**Pull request by lauraguerra1**
Description
<br>
This PR will put all of the fetches into one promise resolution, and dries up the code by separating the actions from the original `.then`'s into separate functions. Those functions are invoked as the data is iterated over within the promise resolution.

Testing Notes
<br>
- Load a page and make sure that no errors pop up as you click on recipes and save them
- Load a page with chatGPT config file information to ensure the extension still works

### CODE REVIEW

File: src/apiCalls.js [hunk 1]
```diff
@@ -13,72 +13,56 @@ let pageData = {
...
-        });
-}
+const handleRecipeData = recipes => {
+  pageData.allRecipes = recipes;
+  pageData.recipesOfInterest = copyItem(pageData.allRecipes);
+  pageData.allTags = populateTags(pageData.allRecipes);
+  // getChatGPTRecipePitches(pageData.allRecipes);
+  setTimeout(() => {
+  hideSpinner();
+  pageLoadRenders(pageData.allRecipes);
```
Comment (mbenfowler) [inline src/apiCalls.js]: indent

File: src/apiCalls.js [hunk 2]
```diff
@@ -109,4 +93,10 @@ const getChatGPTRecipePitches = (allRecipes) => {
...
 }
 
-export { currentUser, pageData, updateCurrentUser, loadData, getRecipeCard };
+export { currentUser, pageData, updateCurrentUser, loadData };
+
+
+
+
+
+
```
Comment (mbenfowler) [inline src/apiCalls.js]: a lot of extra spaces here XD

File: src/domUpdates.js [hunk 3]
```diff
@@ -333,7 +334,7 @@ const displayTaggedRecipes = () => {
   const activeTags = pageData.allTags.filter(tag => tag.isActive).map(tag => tag.name);
   const baseData = setBaseData();
   const filteredRecipes = setupFilterData(activeTags, baseData);
-  
+  console.log('here', filteredRecipes)
```
Comment (mbenfowler) [inline src/domUpdates.js]: remove

File: src/apiCalls.js [hunk 4]
```diff
@@ -13,72 +13,56 @@ let pageData = {
...
+        if(response.ok) {
+          response.json()
+          .then (data => {
+            const functions = {
+              users: (users) => handleUserData(users), 
+              recipes: (recipes) => handleRecipeData(recipes),
+              ingredients: (ingredients) => handleIngredientData(ingredients)
+            }
+            const property = response.url.split('/').reverse()[0]
+            functions[property](data[property])
```
Comment (mbenfowler) [inline src/apiCalls.js:50]: 🔥🔥 🔥

**Review by mbenfowler — CHANGES_REQUESTED** [review]
just minor edits requested

**Comment by mbenfowler** [comment]
[[FEEDBACK] DRY up apiCalls.js](https://trello.com/c/TqsowCLU/48-feedback-dry-up-apicallsjs)

File: src/apiCalls.js (hunk 1 above)
Comment (lauraguerra1) [reply src/apiCalls.js]: done

File: src/apiCalls.js (hunk 2 above)
Comment (lauraguerra1) [reply src/apiCalls.js]: lol had to make sure that I got that extra line at the end of the file!!

Lol jk, done now

File: src/domUpdates.js (hunk 3 above)
Comment (lauraguerra1) [reply src/domUpdates.js]: done!

File: src/apiCalls.js (hunk 2 above)
Comment (mbenfowler) [reply src/apiCalls.js]: LOL hahaha

**Review by mbenfowler — APPROVED** [review]
just minor edits requested

**Review by taranveersingh93 — APPROVED** [review]
Sweet

**Review by taranveersingh93 — APPROVED** [review]
Sweet

**Pull request merged**